src/systems/world_generation/WorldGeneration.ts

```diff
@@ -0,0 +1,171 @@
+import forest_path from "../../assets/environment/tiles/forest.png";
+import grass_path from "../../assets/environment/tiles/grass.png";
+import mountain_path from "../../assets/environment/tiles/mountains.png";
+import ocean_path from "../../assets/environment/tiles/ocean_deep.png";
+import { randomInt } from "../../helpers/math/MathHelper";
+
+// Define enum for tile types
```

Comment:
Most comments we don't need as the code's quite self-explanatory, there's good function names etc